src/components/EnrolmentVillagesPicker.js

```diff
@@ -66,20 +69,77 @@ const EnrolmentVillagesPicker = (props) => {
   const onInsertRow = () => {
     setItems([...items, {}]);
   };
+
   const onSelectParent = (item, parent) => {
-    // eslint-disable-next-line no-param-reassign
-    item.parent = parent;
+    const rowItem = item;
+    rowItem.parent = parent;
     setItems([...items]);
   };
+
   const onVillagesChange = (item, entities) => {
-    // eslint-disable-next-line no-param-reassign
-    item.entities = entities;
+    const rowItem = item;
+    rowItem.entities = entities;
     handleChange(items);
   };
 
-  // Only display options for municipalities not already present.
+  const createRow = (parent) => {
+    const {
+      children: { edges },
+    } = parent;
+    const entities = edges?.map((edge) => edge.node) ?? [];
+    const createdRow = {};
+
+    createdRow.parent = parent;
+    createdRow.entities = entities;
+    return createdRow;
+  };
+
+  const clearItems = () => {
+    setItems([]);
+    handleChange([]);
+  };
+
+  const executeNewRows = (newRows, uniqueRows) => {
+    if (!items.length) {
+      setItems([...items, ...newRows]);
```

Comment:
Could we create separate function for those lines? 104-105 and 107-108 looks really similar.

Reply:
I created new `rescheduleItems` function - it seperates those lines now.